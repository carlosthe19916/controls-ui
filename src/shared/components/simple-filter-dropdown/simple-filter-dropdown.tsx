import React, { useState } from "react";
import {
  Dropdown,
  DropdownPosition,
  DropdownToggle,
  DropdownItem,
} from "@patternfly/react-core";
import { FilterIcon } from "@patternfly/react-icons";

export interface SimpleFilterDropdownProps {
  label: string;
  options: string[];
  onSelect: (value: string) => void;
}

export const SimpleFilterDropdown: React.FC<SimpleFilterDropdownProps> = ({
  label,
  options,
  onSelect,
}) => {
  const [isOpen, setIsOpen] = useState(false);

  const handleOnSelect = () => {
    setIsOpen((current) => !current);
  };

  const handleOnToggle = (isOpen: boolean) => {
    setIsOpen(isOpen);
  };

  return (
    <Dropdown
      position={DropdownPosition.left}
      isOpen={isOpen}
      onSelect={handleOnSelect}
      toggle={
        <DropdownToggle onToggle={handleOnToggle}>
          <FilterIcon />
          {label}
        </DropdownToggle>
      }
      dropdownItems={options.map((f, index) => (
        <DropdownItem
          key={index}
          component="button"
          onClick={() => onSelect(f)}
        >
          {f}
        </DropdownItem>
      ))}
    />
  );
};
