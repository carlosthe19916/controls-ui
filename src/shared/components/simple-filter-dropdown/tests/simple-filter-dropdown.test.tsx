import React from "react";
import { shallow, mount } from "enzyme";
import { SimpleFilterDropdown } from "../simple-filter-dropdown";

describe("SimpleFilterDropdown", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(
      <SimpleFilterDropdown
        label="My label"
        options={["option1", "option2", "option3"]}
        onSelect={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Check onSelect callback", () => {
    const onSelectSpy = jest.fn();

    const wrapper = mount(
      <SimpleFilterDropdown
        label="My label"
        options={["option1", "option2", "option3"]}
        onSelect={onSelectSpy}
      />
    );

    // Open dropdown
    wrapper.find(".pf-c-dropdown__toggle").simulate("click");
    wrapper.update();

    // Select option
    wrapper.find(".pf-c-dropdown__menu-item").at(1).simulate("click");
    expect(onSelectSpy).toHaveBeenCalledWith("option2");
  });
});
