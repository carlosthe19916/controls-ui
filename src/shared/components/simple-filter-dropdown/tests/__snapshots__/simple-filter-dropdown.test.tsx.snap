// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleFilterDropdown Renders without crashing 1`] = `
<Dropdown
  dropdownItems={
    Array [
      <DropdownItem
        component="button"
        onClick={[Function]}
      >
        option1
      </DropdownItem>,
      <DropdownItem
        component="button"
        onClick={[Function]}
      >
        option2
      </DropdownItem>,
      <DropdownItem
        component="button"
        onClick={[Function]}
      >
        option3
      </DropdownItem>,
    ]
  }
  isOpen={false}
  onSelect={[Function]}
  position="left"
  toggle={
    <DropdownToggle
      onToggle={[Function]}
    >
      <FilterIcon
        color="currentColor"
        noVerticalAlign={false}
        size="sm"
      />
      My label
    </DropdownToggle>
  }
/>
`;
